### ISSUE TITLE
Object / Structured output

### ISSUE THREAD

**Issue by sixlive** [echolabsdev/prism#45]

**Comment by slashequip** [comment]
Is this including `strict: true` [for Tools](https://platform.openai.com/docs/guides/function-calling#:~:text=When%20you%20turn,your%20function%20definition) as well?

Just experienced the need for this as it's not adhering to my Enums 🫠

**Comment by ejunker** [comment]
Hey TJ, I recently got caught up on listening to the Slightly Caffeinated podcast and heard you talking about adding structured output to Prism. I am wondering if this includes deserializing the response JSON into PHP objects?

I took a look at how some other PHP libraries do this and here they are. Maybe they will provide some inspiration on how to handle it.

https://github.com/cognesy/instructor-php
- Define response data model the way you want: type-hinted classes, JSON Schema arrays, or dynamic data shapes with Structure class
- Use attributes or PHP DocBlocks to provide additional instructions for LLM
- Customize response model processing by providing your own implementation of schema, deserialization, validation and transformation interfaces

https://github.com/php-llm/llm-chain?tab=readme-ov-file#structured-output
- deserialize JSON response into PHP classes using `symfony/serializer`

I'm really looking forward to Structured Output. I think it will be a very useful feature! My main use case would be taking unstructured data like a candidate's resume and then parse out the address, previous employment, skills, education, etc. into PHP instances of pre-defined classes.
